test/FT-voting-quadratic/RIFGovernorFtQuadratic.test.js

```diff
@@ -171,21 +183,64 @@ describe('Governance - Fungible tokens voting', () => {
             reason,
           );
       });
+    });
 
+    describe('Voting results', () => {
       it('total votes should equal voters rif amount square root sum', async () => {
         const againstVotes = await governor.getAgainstVotes(proposalId);
-        expect(againstVotes).to.equal(
-          Math.sqrt(voters[0].rifAmount) + Math.sqrt(voters[1].rifAmount),
-        );
+        expect(againstVotes).to.equal(Math.sqrt(voters[1].rifAmount));
         const forVotes = await governor.getForVotes(proposalId);
-        expect(forVotes).to.equal(Math.sqrt(voters[2].rifAmount));
-        console.log(forVotes, againstVotes);
+        expect(forVotes).to.equal(
+          Math.sqrt(voters[2].rifAmount) + Math.sqrt(voters[0].rifAmount),
+        );
+      });
+
+      it('should calculate the quorum correctly', async () => {
+        const quorum = await governor.quorum(
+          (await hre.ethers.provider.getBlockNumber()) - 1,
+        );
+        expect(quorum).to.equal(Math.floor(Math.sqrt(totalSupply)));
+      });
+
+      it('quorum should be reached', async () => {
+        expect(await governor.quorumReached(proposalId)).to.be.true;
       });
 
-      // TODO: Test fails. Figure out why
       it('voting should be successfull', async () => {
         expect(await governor.voteSucceeded(proposalId)).to.be.true;
       });
     });
+
+    describe('Proposal execution', () => {
+      it('should execute the Proposal and call its target contract', async () => {
+        const deadline = (
+          await governor.proposalDeadline(proposalId)
+        ).toNumber();
+        const currentBlock = await hre.ethers.provider.getBlockNumber();
+        await skipBlocks(deadline - currentBlock + 1);
+        const tx = governor.execute(
+          [rifVoteToken.address, governor.address],
+          [0, 0],
+          [proposalCalldata, setTargetCalldata],
+          proposalDescriptionHash,
+        );
+        await expect(tx)
+          .to.emit(proposalTarget, 'ProposalProcessed')
+          .withArgs(proposalId);
+      });
+
+      it("governor's RIF treasury tokens should be transferred to the team", async () => {
+        expect(await rifVoteToken.balanceOf(team.address)).to.equal(
+          treasuryRifAmount,
+        );
+        expect(await rifVoteToken.balanceOf(governor.address)).to.equal(0);
+      });
+
+      it('address of the proposal target should be set on the governor', async () => {
+        expect(await governor.proposalTarget()).to.equal(
+          proposalTarget.address,
+        );
+      });
+    });
```

Comment:
So above we have a proposal execution attempt on a proposal which succeeded ("happy path").
We'll need another one where there is a proposal execution attempt on a proposal which failed.

Reply:
This test file might be getting too large however, so perhaps have one file for happy path, and a different one for the failure scenarios.